File: 2296-91-1930-unique-length-3-palindromic-subsequences/2296-91-1930-unique-length-3-palindromic-subsequences.java
import java.util.*;

class Solution {
    public int countPalindromicSubsequence(String s) {
        int[] left = new int[26];
        int[] right = new int[26];
        Arrays.fill(left, -1);

        for (int i = 0; i < s.length(); i++) {
            int idx = s.charAt(i) - 'a';
            if (left[idx] == -1) {
                left[idx] = i;
            } else {
                right[idx] = i;
            }
        }

        int ans = 0;
        for (int i = 0; i < 26; i++) {
            if (right[i] == 0) continue;
            boolean[] visited = new boolean[26];
            for (int j = left[i] + 1; j < right[i]; j++) {
                int idx = s.charAt(j) - 'a';
                if (!visited[idx]) {
                    visited[idx] = true;
                    ans++;
                }
            }
        }

        return ans;
    }
}

File: 2296-91-1930-unique-length-3-palindromic-subsequences/2296-91-1930-unique-length-3-palindromic-subsequences.js
var countPalindromicSubsequence = function(s) {
    const left = Array(26).fill(-1);
    const right = Array(26).fill(0);

    for (let i = 0; i < s.length; i++) {
        const idx = s.charCodeAt(i) - 'a'.charCodeAt(0);
        if (left[idx] === -1) {
            left[idx] = i;
        } else {
            right[idx] = i;
        }
    }

    let ans = 0;
    for (let i = 0; i < 26; i++) {
        if (right[i] === 0) continue;
        const visited = new Set();
        for (let j = left[i] + 1; j < right[i]; j++) {
            visited.add(s[j]);
        }
        ans += visited.size;
    }

    return ans;
};
